### ISSUE TITLE
Metric Logging Rounds to 4 Decimal Places, Causing Loss Values Below 0.00005 to Log as Zero

### ISSUE THREAD

**Issue by shaabhishek** [axolotl-ai-cloud/axolotl#3320]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

When training loss values drop below 4 decimal places (e.g., `0.00005`, `0.000001`), they should be logged with full precision to MLflow so users can monitor convergence accurately.

### Current behaviour

All metrics (including `loss`) are rounded to 4 decimal places in `AxolotlTrainer.log()`. This causes any value below `0.00005` to be logged as `0.0000`, making it impossible to monitor training convergence for any training that achieves very low loss values.

### Steps to reproduce

1. Train any model to very low loss (e.g., a small dataset with many epochs)
2. Configure MLflow or WandB logging
3. Observe that once loss drops below `0.0001`, logged values become `0.0`

The outcome is obvious from the [implementation](https://github.com/axolotl-ai-cloud/axolotl/blob/2a664dc8ad178d2881bf361e33559f19f30be829/src/axolotl/core/trainers/base.py#L616):

```python
logs[key] = round(fn(values).item(), 4)
```

### Config yaml

```yaml
base_model: Qwen/Qwen3-4B-Instruct-2507
# ... standard config ...

# MLflow configuration
mlflow_tracking_uri: databricks  # or any mlflow server
mlflow_experiment_name: my-experiment

# Training will eventually hit very low loss
num_epochs: 40
learning_rate: 1e-5
```

### Possible solution

**Option 1: Remove rounding entirely (recommended)**
```python
logs[key] = fn(values).item()
```

**Option 2: Make precision configurable via environment variable**
```python
METRIC_PRECISION = int(os.getenv("AXOLOTL_METRIC_PRECISION", "8"))
logs[key] = round(fn(values).item(), METRIC_PRECISION)
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11

### axolotl branch-commit

main

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hey, thanks for the report. Is it normal for your loss to reach so low? This is a very very small number. We're balancing the value logged to console, otherwise, it'll be messy. I think a config could be one way of addressing this.
